File: CPP/easy/reverseInteger.cpp
#include <vector>
#include <iostream>

int reverse(int x) {
    double ans = 0;
    if(x < -2147483648 || x > 2147483648)
        return 0;

    while(x != 0){
        ans = ans * 10 + x % 10;
        x /= 10;
    }
    return (ans <= -2147483648 || ans >= 2147483648) ? 0 : ans;
}

int main(){
    std::vector<int> test = {123, -123, 0};
    for(int num : test){
        std::cout<< num << " => " << reverse(num) << std::endl;
    }
    return 0;    
}

File: Java/easy/reverseInteger.java
import java.util.*;

class Solution {
    public static int reverse(int x) {
        if(x >= 0 && x < 10) return x;
        if(x > Integer.MAX_VALUE || x < Integer.MIN_VALUE)  return 0;
        boolean negative = x < 0;

        x = Math.abs(x);

        double ans = 0;

        while(x > 0){
            ans = (ans * 10 + x % 10);
            x /= 10;
        }

        ans = (negative? -ans : ans);
        if(ans > Integer.MAX_VALUE || ans < Integer.MIN_VALUE)  return 0;
        return (int)ans;
    }
    
    // Original = 0    => 0
    // Original = 120  => 21       
    // Original = 123  => 321      
    // Original = -123 => -321     
    // Original = 1534236469   => 0

    public static void main(String[] args){
        int[] testSet = new int[]{0, 120, 123, -123, 1534236469};

        for(int test : testSet){
            System.out.println("Original = " + test + "\t=> " + reverse(test));
        }
    }
}